dspace-api/src/main/java/org/dspace/xmlworkflow/state/actions/userassignment/ClaimAction.java

```diff
@@ -42,6 +46,12 @@ public class ClaimAction extends UserSelectionAction {
     private final ConfigurationService configurationService
             = DSpaceServicesFactory.getInstance().getConfigurationService();
 
+
+    private final XmlWorkflowServiceFactory factory = XmlWorkflowServiceFactory.getInstance();
+
+    private final XmlWorkflowService xmlWorkflowService = factory.getXmlWorkflowService();
+    private final XmlWorkflowFactory workflowFactory = factory.getWorkflowFactory();
```

Comment:
I don't think we should be caching these in global private variables.

  I'm noticing that other areas of this same class are calling them directly instead.  For instance, look at the `execute()` method and `alertUsersOnActivation()` method of `ClaimAction`.  Both of those also use `XmlWorkflowServiceFactory.getInstance()` but only call it when it's needed.

I'd recommend we keep the same approach for this new PR & remove these three global variables.